File: src/Domain/Entities/Forum.cs
using Forum.Domain.Common;
using Forum.Domain.Enums;

namespace Forum.Domain.Entities;

public class ForumEntity : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public ForumElementType ElementType { get; } = ForumElementType.Forum;
    public int? ParentForumId { get; set; }
    public string? Category { get; set; }
    public string? Description { get; set; }
    public IQueryable<ForumEntity>? Subforums { get; set; }
    public IQueryable<Topic>? Topics { get; set; }
}

File: src/WebUI/src/app/common/models/forum.ts
import { Category } from "./category";
import { Subforum } from "./subforum";
import { TopicInForum } from "./topic-in-forum";

export interface Forum {
    id: number,
    name: string,
    parentForumId: number | null,
    category: string | null,
    description: string | null,
    isClosed: boolean,
    isUnread: boolean,
    subcategories: Category[],
    subforums: Subforum[],
    topics: TopicInForum[]
}